lib/ASRouter.jsm

```diff
@@ -55,21 +55,45 @@ const MessageLoaderUtils = {
    *
    * @param {obj} provider An AS router provider
    * @param {string} provider.url An endpoint that returns an array of messages as JSON
+   * @param {obj} storage A storage object with get() and set() methods for caching.
    * @returns {Promise} resolves with an array of messages, or an empty array if none could be fetched
    */
-  async _remoteLoader(provider) {
+  async _remoteLoader(provider, storage) {
     let remoteMessages = [];
     if (provider.url) {
+      const cacheKey = `RemoteLoaderCache-${provider.url}`;
```

Comment:
I think it would be better to store everything under one cache key, just because clearing it will be easier; also, let's just cache by `provider.id` (since we want to clear outdated responses anyway).

Also, I think it would be good to store the `STARTPAGE_VERSION` (which is really just the version of ASRouter, it's very badly named) so we can invalidate if the major version changes.

I ended up [writing some code](https://github.com/rlr/activity-streams/compare/bug1433243/asr-cache...k88hudson:suggestions-for-caching) to think about this one if it's helpful

Reply:
that makes sense, awesome! thanks for the patch 🕺